microsoft/testsuites/gpu/gpusuite.py

```diff
@@ -0,0 +1,138 @@
+from lisa.sut_orchestrator.azure.tools import LisDriver
+from lisa.testsuite import simple_requirement
+from lisa.tools import Lscpu, Lsmod
+from lisa.util import LisaException, SkippedException
+from lisa.util.perf_timer import create_timer
+
+
+@TestSuiteMetadata(
+    area="gpu",
+    category="functional",
+    description="""
+    This test suite runs the gpu test cases.
+    """,
+)
+class gpu(TestSuite):
+    grid_supported_skus = ["Standard_NV"]
+    cuda_supported_skus = ["Standard_NC", "Standard_ND"]
+
+    def _validate_node_reboot(self, node: Node, case_path: Path) -> None:
+        serial_console = node.features[SerialConsole]
+        try:
+            timer = create_timer()
+            node.reboot()
+            self.log.debug(f"node '{node.name}' rebooted in {timer}")
+        except Exception as identifier:
+            # if there is any panic, fail before partial pass
+            serial_console.check_panic(saved_path=case_path, stage="reboot")
+
+            # if node cannot be connected after reboot, it should be failed.
+            if isinstance(identifier, LisaException) and str(identifier).startswith(
+                "cannot connect to TCP port"
+            ):
+                raise LisaException(f"after reboot, {identifier}")
+
+    def _get_expected_device_count(self, node: Node) -> int:
+        lscpu = node.tools[Lscpu]
+        core_count = lscpu.get_core_count()
+        expected_count = math.ceil(core_count / 6)
+
+        node_runbook = node.capability.get_extended_runbook(AzureNodeSchema, AZURE)
+        if "Standard_ND96asr" in node_runbook.vm_size:
+            expected_count = math.ceil(core_count / 12)
+        elif "Standard_NDv2" in node_runbook.vm_size:
+            expected_count = math.ceil(core_count / 5)
+        elif (
+            "Standard_ND" in node_runbook.vm_size
+            or "Standard_NV" in node_runbook.vm_size
+        ) and "v3" in node_runbook.vm_size:
+            expected_count = math.ceil(core_count / 12)
+
+        return expected_count
+
+    @TestCaseMetadata(
+        description="""
+            This test case verfies kernel support for gpu.
+            TODO: validate the test params against GPU support test matrix
+                    when test matrix is available.
+        """,
+        requirement=simple_requirement(
+            supported_features=[Gpu],
+        ),
+    )
+    def validate_gpu_support(self, case_name: str, node: Node) -> None:
+        gpu_feature = node.features[Gpu]
+        gpu_feature.check_support()
+
+    @TestCaseMetadata(
+        description="""
+            This test case verfies if gpu drivers are loaded fine.
+
+            Steps:
+            1. Validate if the VM SKU is supported for GPU.
+            2. Install LIS drivers if not already installed for Fedora and its
+                derived distros. Reboot the node
+            3. Install required gpu drivers on the VM and reboot the node. Validate gpu
```

Comment:
Do we really need to reboot twice or do we defer to one reboot? Also, reboots needed at all?

Reply:
reboot needed once. but yeah let me make it once instead of twice.